[Security Solution] Implement rule monitoring dashboard

banderror opened the pull request:
**Epic:** https://github.com/elastic/security-team/issues/6032 (internal)

## Summary

This PR adds a new `[Elastic Security] Detection rule monitoring` Kibana dashboard and a new `POST /internal/detection_engine/health/_setup` API endpoint.

## Dashboard

The dashboard can be helpful for monitoring the health and performance of Security detection rules. Users of the dashboard must have read access to the `.kibana-event-log-*` index (see below for exact requirements). The dashboard is automatically installed into the current Kibana space when a user visits a page in Security Solution - similar to how we install the Fleet package with prebuilt detection rules.

<img width="1791" alt="Kibana dashboards page" src="https://github.com/elastic/kibana/assets/7359339/92cb3c75-39ea-4069-b70f-8f531869edf7">

<img width="1775" alt="Security dashboards page" src="https://github.com/elastic/kibana/assets/7359339/3b27aeb6-2222-40fd-a453-c204fcee4f31">

![Rule monitoring dashboard itself](https://github.com/elastic/kibana/assets/7359339/755cc044-5613-4c78-b89f-2a9734ded76d)


## API endpoint

The PR also adds a new endpoint for setting up anything related to monitoring rules and the health of the Detection Engine. If you call the endpoint, it will install the new dashboard to the Default Kibana space:

```
POST /internal/detection_engine/health/_setup
```

In order to install the dashboard to a different Kibana space, you will need to call it like that:

```
POST /s/<space-id>/internal/detection_engine/health/_setup
```

## RBAC requirements

### For installing the dashboard

The user calling the `POST /internal/detection_engine/health/_setup` endpoint must have at least `Read` access to Security Solution. No additional privileges are required, because the endpoint installs the dashboard on behalf of the internal user (`kibana_system`).

<img width="739" alt="Screenshot 2023-06-22 at 21 00 58" src="https://github.com/elastic/kibana/assets/7359339/b622bc40-9c8c-4e2c-a9e7-6c3d6b8e6589">

### For viewing the dashboard

In order to be able to view the dashboard, users will need to have at least the following 2 privileges:

1. At least `Read` access to Kibana dashboards (`Analytics` -> `Dashboard` : `Read`)

<img width="731" alt="Screenshot 2023-06-22 at 20 49 38" src="https://github.com/elastic/security-docs/assets/7359339/e7f52c29-0492-423e-8fb2-0dab9b5bd400">

2. At least `read` access to the `.kibana-event-log-*` index.

<img width="1472" alt="Screenshot 2023-06-22 at 20 52 10" src="https://github.com/elastic/security-docs/assets/7359339/844f650b-b285-48ae-ac59-42831fc406d9">

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
  - [x] https://github.com/elastic/security-docs/issues/3478
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/security-detections-response (Team:Detections and Resp)

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

jpdjere reviewed (approved):
Installed successfully, tested and working as described 👍 

Are we not planning on adding this dashboard to the Security Dashboards page?
It's hard to find through the general dashboards list.

approksiu commented:
> Installed successfully, tested and working as described 👍
> 
> Are we not planning on adding this dashboard to the Security Dashboards page? It's hard to find through the general dashboards list.

We should add it there - by adding "Security Solution" tag. cc @banderror

banderror commented:
Based on the received feedback I'm starting to work on the following fixes and improvements:

- [ ] Docs
  - [ ] Open a ticket for documenting the dashboard in the user-facing docs
  - [ ] Known issue should be added to the release notes: rule can write multiple final statuses per execution; can be often caused by gap detection
- [ ] Fix the dashboard itself
  - [ ] Change the "top 5 rules" label to "top 5 rules per interval"
  - [ ] Add descriptions to all the panels
  - [ ] Fix the 3 tables at the bottom by adding columns properly
  - [ ] Add the `Managed` and `Security Solution` tags to the dashboard as it was done for the `Threat Intelligence Overview` dashboard
- [ ] Fix the way the dashboard gets installed
  - [ ] Install it on behalf of the internal user
  - [ ] Remove the installation on Kibana server start
  - [ ] Call the `_setup` endpoint from all Security Solution pages, similar to how we install the package with prebuilt rules

banderror commented:
@approksiu @jpdjere I addressed all the [feedback](https://github.com/elastic/kibana/pull/159875#issuecomment-1597489157) and updated the PR description accordingly. Hopefully, the CI will pass and the PR will be merged soon.

banderror commented:
Thanks for the suggestion @philippkahr, I think this could be a nice usability enhancement. Would you mind opening an issue in https://github.com/elastic/kibana for that?
